### PR TITLE
Mod version checker

### PR DESCRIPTION
**Pull request by VirxEC**
If you accept this, your mod will:

1. Check if another version of this mod has been released, and notify the user.
2. If the user chooses to, they will be direct to a download link for your mod. (I set it to your download page on CurseForge.)

Right now update.json is incomplete because idk what your changelog is.

All you have to do is change the version of your mod in Reference.java and update update.json (that file should be in the mod; should be on GitHub only) and then that's it! Your users will get a notification for your new version. Example? Download an old version of my mod at https://minecraft.curseforge.com/projects/rubies-mod then go in the game and next to mods you will see a green "emerald?" (idk what it is, Forge is weird), click on it, go to the mod, (should have the same symbol next to it) and you will see the changelog and update URL.

### CODE REVIEW

**Comment by steinm07** [comment]
Looks promising.

**Comment by suppergerrie2** [comment]
Did you test if the mod works in these versions? Idk if something changed or if the mod uses methods that are only in newer versions?

File: src/main/java/com/suppergerrie2/sdrones/Reference.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 	public static final String MODID = "sdrones";
 	public static final String MODNAME = "Suppergerrie2's Drone Mod";
 	public static final String VERSION = "1.3.1";
-	public static final String ACCEPTED_MINECRAFT_VERSIONS = "[1.12]";
+	public static final String ACCEPTED_MINECRAFT_VERSIONS = "[1.12][1.12.1][1.12.2]";
```
Comment (suppergerrie2) [inline src/main/java/com/suppergerrie2/sdrones/Reference.java]: This doesnt look correct, check this site for the syntax:
https://cwiki.apache.org/confluence/display/MAVENOLD/Dependency+Mediation+and+Conflict+Resolution#DependencyMediationandConflictResolution-DependencyVersionRanges

File: update.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+	"homepage": "https://minecraft.curseforge.com/projects/suppergerrie2s-drone-mod",
+	"1.12.2": {
+		"1.3.1": "changelog here\nsupports special character like \\n"
```
Comment (suppergerrie2) [inline update.json]: You can find the changelog here:
https://minecraft.curseforge.com/projects/suppergerrie2s-drone-mod/files/2607992

File: src/main/java/com/suppergerrie2/sdrones/Reference.java (hunk 1 above)
Comment (VirxEpicoder) [reply src/main/java/com/suppergerrie2/sdrones/Reference.java]: Fixed this. Sorry!

File: update.json (hunk 2 above)
Comment (VirxEpicoder) [reply update.json]: Added your changelog.

**Comment by VirxEpicoder** [comment]
Fixed all recommendation please review again @suppergerrie2

**Comment by VirxEpicoder** [comment]
Sorry for the latency but yes ur mod works in the versions, as they are extremely similar except for some bug fixes on Mojang's part @suppergerrie2

**Comment by suppergerrie2** [comment]
I'm more worried about forge changes :D

**Comment by VirxEpicoder** [comment]
I can remove it if you want

**Comment by VirxEpicoder** [comment]
Fixed it @suppergerrie2 it now only runs in 1.12

**Comment by suppergerrie2** [comment]
Was just saying why I asked if you had tested it. :P

**Comment by VirxEpicoder** [comment]
No, but if you accept this (1.12 only) I will create another pull request later to update it to 1.12.1 and 1.12.2 and make sure that it works.

**Comment by suppergerrie2** [comment]
You should probably set it to 1.12.2 ATM, because that is the version it is developed on.

**Comment by VirxEpicoder** [comment]
got it

**Comment by VirxEpicoder** [comment]
Done

**Comment by VirxEpicoder** [comment]
@suppergerrie2 can confirm that ur mod doesn't work in 1.12.1 Forge; errors with EntityAIAttackRanged, EntityAISearchItems, EntityArcherDrone, RecipeFilterDrone, and ShapedDroneUpgradeRecipe. The 1.12.1 and 1.12 compatibility has been removed from this pull request, and will now only work in 1.12.2 Forge.

**Comment by VirxEpicoder** [comment]
Can you please accept this pull request now?

File: update.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+	"homepage": "https://minecraft.curseforge.com/projects/suppergerrie2s-drone-mod",
+	"1.12.2": {
+		"1.3.1": "Fixed drone not dropping spawn item when it was spawned with the last item of the stack.\nFixed crop farm drone not farming or breaking blocks when out of dirt.\nFixed drone_upgrade not in lang file."
+	},
+	"promos": {
+		"1.12-latest": "1.3.1",
+		"1.12-recommended": "1.3.1"
```
Comment (suppergerrie2) [inline update.json]: Should these be 1.12.2 instead of 1.12?

Comment (VirxEC) [reply update.json]: Fixed :)

**Pull request merged**